app/controllers/api/v2/hostgroups_controller.rb

```diff
@@ -94,12 +94,32 @@ def clone
         process_response @hostgroup.save
       end
 
+      api :PUT, "/hostgroups/:id/rebuild_config", N_("Rebuild orchestration config")
+      param :id, :identifier, :required => true
+      param :only, Array, :desc => N_("Limit rebuild steps, valid steps are %{host_rebuild_steps}")
+      param :children_hosts, :bool, :desc => N_("Operate on child hostgroup hosts")
+      def rebuild_config
+        results = @hostgroup.recreate_hosts_config(params[:only], params[:children_hosts])
+        failures = []
+        results.each_pair do |host, result|
+          host_failures = result.reject { |key, value| value }.keys.map{ |k| _(k) }
+          failures << "#{host}(#{host_failures.to_sentence})" unless host_failures.empty?
+        end
+        if failures.empty?
+          render_message _("Configuration successfully rebuilt."), :status => :ok
+        else
+          render_message (_("Configuration rebuild failed for: %s." % failures.to_sentence)), :status => :unprocessable_entity
```

Comment:
You can use the standard `render_error` function, to render the error, like [here](https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/hosts_controller.rb#L156).